src/state/transactions/updater.tsx

```diff
@@ -41,14 +42,46 @@ export default function Updater(): null {
 
     useEffect(() => {
         if (!chainId || !library || !lastBlockNumber) return
-
         Object.keys(transactions)
             .filter(hash => shouldCheck(lastBlockNumber, transactions[hash]))
             .forEach(hash => {
                 library
                     .getTransactionReceipt(hash)
                     .then(receipt => {
+                      let confirmedSummary;
                         if (receipt) {
+                          if (transactions[hash]?.summary) {
+                            const receiptData = receipt.logs[receipt.logs.length - 1].data
+                            const txInput = transactions[hash]?.tradeInfo?.input
+                            const txOutput = transactions[hash]?.tradeInfo?.output
+                            if (txInput?.symbol !== 'G$'){
+                              // Buying G$
+
+                              const dataDecodeBuy = utils.defaultAbiCoder.decode(['uint256', 'uint256'], receiptData)
+                              const inputAmount = utils.formatUnits(
+                                  dataDecodeBuy[0], txInput?.decimals 
+                              )
+                              const fixedInput = FixedNumber.fromString(inputAmount).round(5)
+                              const actualReturnAmount = utils.commify(utils.formatUnits(dataDecodeBuy[1], 2))
+                              confirmedSummary = 'Swapped ' + fixedInput + ' ' 
+                              + txInput?.symbol + 
+                              ' to ' + actualReturnAmount + ' ' +
+                              txOutput?.symbol
+                            } else {
+                              // Selling G$
+                              const dataDecodeSell = utils.defaultAbiCoder.decode(['uint256', 'uint256', 'uint256'], receiptData)
+                              const inputAmount = utils.commify(utils.formatUnits(
+                                dataDecodeSell[0], txInput?.decimals
+                              ))
+                              const actualReturnAmount = utils.formatUnits(dataDecodeSell[2], txOutput?.decimals)
+                              const fixedReturn = FixedNumber.fromString(actualReturnAmount).round(5)
+                              console.log({fixedReturn}) 
+                              confirmedSummary = 'Swapped ' + inputAmount + ' '
+                              + txInput?.symbol + 
+                              ' to ' + fixedReturn + ' ' +
+                              txOutput?.symbol
```

Comment:
1. how does uniswap/sushiswap worked before this addition?
2. can this code be more DRY? i dont see much difference between buy/sell "Swapped X for Y"

Reply:
1. Sushi uses several custom functions/classes from sdk-core, which we don't have...

the most basic explanation I can give for now (and as far as I understand) 
is that they actually determine what the output will be so from transaction send they already have a summary with calculated output, and they never update after

So they have (trade || tridentTrade) which use their Trade Class (and some parse functions) 
to end up with this (when adding transaction) >

        if (tridentTradeContext?.parsedAmounts) {
        base = `Swap ${tridentTradeContext?.parsedAmounts[0]?.toSignificant(4)} ${
                        tridentTradeContext?.parsedAmounts[0].currency?.symbol
                     ` 
        } for ${tridentTradeContext?.parsedAmounts[1]?.toSignificant(4)} ${
                        tridentTradeContext?.parsedAmounts[1].currency?.symbol
                      }
              

So what I believe to have done is this >
to have a summary during pending

    const inputSig = meta?.inputAmount.toSignificant(5)
    const minimumOutputSig = meta?.minimumOutputAmount.toSignificant(5)`

            const tradeInfo = {
              input: {
                decimals: meta?.inputAmount.currency.decimals,
                symbol: meta?.inputAmount.currency.symbol
              },
              output: {
                decimals: meta?.outputAmount.currency.decimals,
                symbol: meta?.outputAmount.currency.symbol
              }
            }
            const summary = 'Swapping ' + inputSig + ' ' 
                            + meta?.inputAmount.currency.symbol + 
                            ' to a minimum of ' + minimumOutputSig + ' ' +
                            meta?.outputAmount.currency.symbol`
                            
to then in updater, when the transaction is confirmed
update summary from receipt data

(attempted Dry'd)

        let decoded, format, commify;
        const symbol = txInput?.symbol !== 'G$'
        if (symbol){
          // Buying G$
          decoded = utils.defaultAbiCoder.decode(['uint256', 'uint256'], receiptData)
        } else {
          // Selling G$
          decoded = utils.defaultAbiCoder.decode(['uint256', 'uint256', 'uint256'], receiptData)
        }

        format = FixedNumber.fromString( utils.formatUnits(
            decoded[symbol ? 0 : 2], symbol ? txInput?.decimals : txOutput?.decimals))
            
        commify = utils.commify(utils.formatUnits(decoded[symbol ? 1 : 0], 
                                symbol ? txOutput?.decimals : txInput?.decimals))

        confirmedSummary = 'Swapped ' + symbol ? format : commify + ' ' 
        + txInput?.symbol + 
        ' to ' + symbol ? commify : format + ' ' +
        txOutput?.symbol
 
 where I use TradeInfo to know input/output currency
 
 and what I now believe to have done also is skip all the logic involved for trying to calculate the trade (with/without) slippage and come up with the actual Return Amount up-front